pennylane/measurements/shots.py

```diff
@@ -175,6 +175,18 @@ def __repr__(self):
         """The representation of the class"""
         return f"Shots(total_shots={self.total_shots}, shot_vector={self.shot_vector})"
 
+    def __eq__(self, other):
+        """Equality between Shot instances."""
+        return self.total_shots == other.total_shots and set(self.shot_vector) == set(
+            other.shot_vector
+        )
+
+    def __hash__(self):
+        """Hash for a given Shot instance."""
+        return hash(
+            (self.total_shots, set(self.shot_vector))  # hash the tuple of these two attributes
```

Comment:
```suggestion
            (self.total_shots, self.shot_vector)  # hash the tuple of these two attributes
```

```
>>> hash(Shots((1,2,3)).shot_vector )
8035285099260388755
>>> hash(Shots((1,2,3)).shot_vector )
8035285099260388755
```
A tuple of named tuples is already immutable and hashable.

Reply:
Added 👍🏼